### DISCUSSION TITLE
Remove HP -> Fail at <75% accuracy

### DISCUSSION THREAD

**Discussion by MangoDragon** [ppy/osu#24299] [Ideas]
I think if the HP concept was removed, this would be more friendly with players. No more weird HP drain, HP bars or choking at the end of a map. Instead, if accuracy falls below 75% at the end of the map, count the score as failed (not added to leaderboards).

Pros:
- Some maps have very abusive HP drain, so this will remove that issue.
- Less UI elements.
- lazer's HP is currently pretty forgiving, so 8* maps can be passed even with very low accuracy. Removing HP means that these low accurcy passes won't be allowed (so that for example, the 8* medal is an actual achievement again).
- Means multiplayer and solo works the same way.
- Allows players to practice the whole song even if they aren't good at one section, without the need for the "no fail" mod.
- Allows beginners to not feel annoyed when they are learning the game because the map won't suddenly end.
- All songs have the same pass criteria (of 75%).

Cons:
- HP might currently be implemented into the PP calculation and might need to be changed, but I'm not sure.
- The current system has been this way for a long time, so some players might not be used to the change.

I think it would be interesting to discuss any more pros/cons or ideas for this change below.

**Comment by peppy** [comment]
> lazer's HP is currently pretty forgiving
the old style is returning

what you are asking for could be implemented as a setting for "Accuracy Challenge" mod, which currently adds an additional fail condition, but a setting could be something like "Don't fail from HP drain" in addition.

**Reply by MangoDragon** [reply]
Actually, the old HP style is very strict in my opinion haha. Especially for tech maps.

That would add this behaviour so that's cool, but mainly what I'm thinking about, is that the fail behaviour should be the same for both multiplayer and solo.

**Reply by peppy** [reply]
it's already the same in multiplayer, assuming you use the same mods.

**Reply by MangoDragon** [reply]
Sorry, I meant like failing doesn't end the song in multiplayer, but it does in solo. I thought it would be nice if it also doesn't end it in solo too, so I came up with this idea.

**Comment by marioman63** [comment]
> Some maps have very abusive HP drain, so this will remove that issue.

This is a part of the map (signified by the HP stat on the map info screen). It is up to the mapper to set this, so it is not so much an issue with the game as a potential balance issue when designing a map. As a result, it can allow for unique types of challenges.

> Less UI elements.

You can disable parts of skins with Lazer's skinning features. You can also just toggle off the HUD altogether. Both are valid solutions to this point

> lazer's HP is currently pretty forgiving, so 8* maps can be passed even with very low accuracy. Removing HP means that these low accurcy passes won't be allowed (so that for example, the 8* medal is an actual achievement again).

Mapping has changed drastically. again, HP is a setting when making maps, so this isn't inherent to the game itself. I have played both forgiving 8 star maps and 8 star maps where if you miss one note, you spend the entire map playing perfectly to maybe get above 50% HP

> Means multiplayer and solo works the same way.

I'm not sure I agree with this as a positive. There are mods you can enable to accomplish this in single player

> Allows players to practice the whole song even if they aren't good at one section, without the need for the "no fail" mod.

Again, disagree with the notion. No Fail is meant for practice.                                                                                                                                                                        

> Allows beginners to not feel annoyed when they are learning the game because the map won't suddenly end.

"Dying" is an intrinsic part of many rhythm games. part of the challenge is meeting a certain threshold of skill to "survive" the map. I don't think you can simply remove it with the notion of "fixing" this "issue"

> All songs have the same pass criteria (of 75%).

This could significantly affect the score ranges of successful plays. What exactly would this accomplish?

**Comment by Rekunan** [comment]
> HP might currently be implemented into the PP calculation and might need to be changed, but I'm not sure.

You'll be glad to hear that most, if not all, of the pp people also dislike HP and failing, so this will never happen.
